event_routing_backends/models.py

```diff
@@ -40,6 +43,80 @@ def get_value_from_dotted_path(dict_obj, dotted_key):
     return result
 
 
+class RouterConfigurationModelManager(models.Manager):
+    """
+    Query manager for ConfigurationModel
+    """
+    def _current_ids_subquery(self):
+        """
+        Internal helper method to return an SQL string that will get the IDs of
+        all the current entries (i.e. the most recent entry for each unique set
+        of key values). Only useful if KEY_FIELDS is set.
+        """
+        return self.values(*self.model.KEY_FIELDS).annotate(max=models.Max('pk')).values('max')
+
+    def current_set(self):
+        """
+        A queryset for the active configuration entries only. Only useful if KEY_FIELDS is set.
+        Active means the means recent entries for each unique combination of keys. It does not
+        necessaryily mean enbled.
+        """
+        assert self.model.KEY_FIELDS != (), "Just use model.current() if there are no KEY_FIELDS"
+        return self.get_queryset().filter(
+            pk__in=self._current_ids_subquery()
+        ).annotate(
+            is_active=models.Value(1, output_field=models.IntegerField())
+        )
+
+    def get_routers(self, backend_name):
+        """
+        A queryset for the active configuration entries only. Only useful if backend_name is passed.
+        This function will return all active routers of a backend.
+        """
+        assert backend_name != '', "Just use model.current() if there are no backend passed"
+
+        cache_key = self.cache_key_name(backend_name)
+        cached_response = TieredCache.get_cached_response(cache_key)
+        if cached_response.is_found:
+            return cached_response.value
+
+        filter_key_dict = {'backend_name': backend_name}
+        current = self.current_set().filter(**filter_key_dict).order_by('-change_date')
```

Comment:
I think we have a lot of unnecessary code in this manager. All we need is a method like `get_routers` and that method should return all enabled routers of a backend from cache if they are in cache if they are not in cache we should cache them before returning. Here is the queryset
`self.get_queryset().filter(backend_name=backend_name, enabled=True)`

Reply:
self.current_set().filter(backend_name=backend_name, enabled=True).order_by('-change_date')
As on update we are creating a new record for config in edx. So I have used above query to get current active records